### PR TITLE
[Core][AttrList] Support creating AttrList with value

### PR DESCRIPTION
**Pull request by ClydeZhao**
ref #119
Usage:
```
objlist.create_attrlist<AttrT>("attr_name", AttrT(...));
```

### CODE REVIEW

**Comment by ClydeZhao** [comment]
@benben1026 Please have a look.

File: core/objlist.hpp [hunk 1]
```diff
@@ -208,6 +208,19 @@ class ObjList : public ObjListBase {
         return (*attrlist);
     }
 
+    template<typename AttrT>
+    AttrList<ObjT, AttrT>& create_attrlist(const std::string& attr_name, AttrT default_attr) {
+        if (attrlist_map.find(attr_name) != attrlist_map.end()) {
+            throw base::HuskyException("ObjList<T>::create_attrlist error: name already exists");
+        }
+        auto* attrlist = new AttrList<ObjT, AttrT>(&objlist_data_);
```
Comment (Yuzhen11) [inline core/objlist.hpp]: Looks like that you still need to use the default constructor. I think we should consider if the AttrT doesn't have default constructor.

Maybe we should provide constructor for AttrList like `AttrList<ObjT, AttrT>(&objlist_data_, default_attr)`.

You can also refer to the constructor of `std::vector`.

File: core/objlist.hpp [hunk 2]
```diff
@@ -208,6 +208,19 @@ class ObjList : public ObjListBase {
         return (*attrlist);
     }
 
+    template<typename AttrT>
+    AttrList<ObjT, AttrT>& create_attrlist(const std::string& attr_name, AttrT default_attr) {
```
Comment (Yuzhen11) [inline core/objlist.hpp]: I will prefer `const AttrT& default_attr`.

**Comment by ClydeZhao** [comment]
@benben1026 @Yuzhen11 This should close the issue. 
If AttrList is created with a user-defined value, e.g., `attrlist = objlist.create_attrlist<AttrT>("attr_name", AttrT(...));`, then when an new object is added to `objlist`, an `AttrT` object initialized with the user-defined value will be added to `attrlist`.

**Comment by Yuzhen11** [comment]
Looks ok.

**Comment by kygx-legend** [comment]
Fine for me. And could you put your information in `CREDITS` file?

**Comment by ClydeZhao** [comment]
@kygx-legend Sure, it's done.

**Pull request merged**